src/libPMacc/include/memory/buffers/HostBufferIntern.hpp

```diff
@@ -101,16 +102,36 @@ class HostBufferIntern : public HostBuffer<TYPE, DIM>
         __startOperation(ITask::TASK_HOST);
         this->setCurrentSize(this->getDataSpace().productOfComponents());
         if (!preserveData)
-            memset(pointer, 0, this->getDataSpace().productOfComponents() * sizeof (TYPE));
+        {
+            /* if it is a pointer out of other memory we can not assume that
+             * that the physical memory is contiguous
+             */
+            if(ownPointer)
+                memset(pointer, 0, this->getDataSpace().productOfComponents() * sizeof (TYPE));
+            else
+            {
+                TYPE value;
+                uint8_t* valuePtr = (uint8_t*)&value;
+                for( size_t b = 0; b < sizeof(TYPE); ++b)
+                {
+                    valuePtr[b] = static_cast<uint8_t>(0);
+                }
+                setValue(value);
```

Comment:
why is a simple `setValue` for the specific type not enough? (needs also a comment)